Include 2.414.1 in SECURITY-3188 list of releases with fix

MarkEWaite opened the pull request:
## Include 2.414.1 in SECURITY-3188 list of releases with fix

Can already be inferred by the reader since the preceding LTS, 2.401.3 includes the fix.  Adding the text makes it clear that the issue does not affect LTS 2.414.1.

May want to consider removing the security advisory item from the 2.414.1 changelog.  It was a backport, but the backport did not result in a user-visible change compared to 2.401.3.

@daniel-beck can you guide @kmartens27 and me on the preferred way to handle the documentation of the security fix backport?  I recommended that Kevin include it in the changelog because it was a backport, but now that I think more deeply, I am not sure that was the correct recommendation.

daniel-beck commented:
> I recommended that Kevin include it in the changelog because it was a backport, but now that I think more deeply, I am not sure that was the correct recommendation.

It's the correct thing to do. You can see us doing this regularly(ish) for when we deliver fixes in 2.OLD.4 / 2.NEW.1.

My only suggestion would be to indicate the date of the advisory when it is not the same date as the release. See e.g. https://www.jenkins.io/changelog/#v2.400, https://www.jenkins.io/changelog-stable/#v2.401.1, https://www.jenkins.io/changelog-stable/#v2.361.1 for previous occurrences.

daniel-beck commented on content/_data/changelogs/lts.yml:
```diff
@@ -9312,7 +9312,7 @@
     references:
       - url: /security/advisory/2023-07-26/
-        title: Security advisory
+        title: 2023-07-26 Security advisory
```
```suggestion
        title: 2023-07-26 security advisory
```

Consistency

daniel-beck reviewed (changes requested):
Top-level metadata `core.lts.fixed` should be changed to `2.401.3 or 2.414.1` so it looks comparable to https://www.jenkins.io/security/advisory/2023-03-08/#fix.

daniel-beck commented on content/security/advisory/2023-07-26.adoc:
Styleguide has nothing on Oxford comma or not 😦

MarkEWaite replied:
I left out the Oxford comma because one of the preceding paragraphs omitted the Oxford comma.  I'm fine either way

daniel-beck replied:
To clarify, do you mean

> Jenkins 2.415 and earlier, LTS 2.401.2 and earlier

?

kmartens27 replied:
I would add the `,` after LTS 2.401.3, since the Oxford comma is typically used in this scenario.

MarkEWaite replied:
> To clarify, do you mean
> 
> > Jenkins 2.415 and earlier, LTS 2.401.2 and earlier
> 
> ?

Yes, that was what I meant.

Now that I read it more carefully, I was misinterpreting.  The sentence I thought was not using an Oxford comma is "A and B" where the phrases "A" and "B" both include the word "and".  That is not a place where an Oxford comma would be used.  My mistake for not reading carefully enough.

daniel-beck replied:
> I would add the `,` after LTS 2.401.3, since the Oxford comma is typically used in this scenario.

I think you're reading the versions wrong in that case. Semantically it is meant to nest as follows:
```
      2.415
    and
      earlier
  ,
    and
      LTS 2.401.2
      earlier

kmartens27 reviewed (approved):
That makes a ton of sense, thanks very much @daniel-beck for clarifying!

MarkEWaite commented:
> Top-level metadata `core.lts.fixed` should be changed to `2.401.3 or 2.414.1` so it looks comparable to https://www.jenkins.io/security/advisory/2023-03-08/#fix.

Fixed in 6a3de943039dd367942a631eea4e99f9bbcfaded

daniel-beck reviewed (approved):
Thanks a lot for taking care of this!

daniel-beck commented on content/security/advisory/2023-07-26.adoc:
Since a change of this kind came up in https://github.com/jenkins-infra/jenkins.io/pull/7036/files#r1466168282, noting that the commit message does not make sense for this change: The previous occurrence is "does not encode".
